### ISSUE TITLE
💅 `correctness/useImportExtensions` not work with path aliases

### ISSUE THREAD

**Issue by danilkinkin** [biomejs/biome#4470]
### Environment information

```bash
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  true
  GraphQL enabled:              false
  Recommended:                  false
  All:                          false
  Enabled rules:
  correctness/useImportExtensions

Workspace:
  Open Documents:               0
```

### Rule name

correctness/useImportExtensions

### Playground link

https://github.com/danilkinkin/biome-correctness-useImportExtensions-bug

### Expected result

Files imported via path aliases also should process by `correctness/useImportExtensions` rule

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
